contracts/REXMarket.sol

```diff
@@ -324,37 +330,49 @@ abstract contract REXMarket is Ownable, SuperAppBase, Initializable {
     }
 
     function updateTokenPrice(ISuperToken _token) public {
-        (
-            bool _ifRetrieve,
-            uint256 _value,
-            uint256 _timestampRetrieved
-        ) = getCurrentValue(market.oracles[_token].requestId);
+        // Optimizing Uniswap V3 consult() code for gas efficiency as we dont need code for harmonicMeanLiquidity
+        // Copied from Uniswap V3 consult(), but removed the harmonicMeanLiquidity code
+        uint32[] memory secondsAgos = new uint32[](2);
+        secondsAgos[0] = secondsAgo;
+        secondsAgos[1] = 0;
+
+        // int56 since tick * time = int24 * uint32
+        // 56 = 24 + 32
+        (int56[] memory tickCumulatives, ) = IUniswapV3Pool(market.oracles[_token].pool).observe(
+            secondsAgos
+        );
 
-        require(_ifRetrieve, "!getCurrentValue");
-        require(_timestampRetrieved >= block.timestamp - 3600, "!currentValue");
+        int56 tickCumulativesDelta = tickCumulatives[1] - tickCumulatives[0];
+
+        // int56 / uint32 = int24
+        int24 tick = int24(tickCumulativesDelta / secondsAgo);
+        // Always round to negative infinity
+        /*
+            int doesn't round down when it is negative
+            int56 a = -3
+            -3 / 10 = -3.3333... so round down to -4
+            but we get
+            a / 10 = -3
+            so if tickCumulativeDelta < 0 and division has remainder, then round
+            down
+        */
+        if (
+            tickCumulativesDelta < 0 && (tickCumulativesDelta % secondsAgo != 0)
+        ) {
+            tick--;
+        }
 
-        market.oracles[_token].usdPrice = _value;
-        market.oracles[_token].lastUpdatedAt = _timestampRetrieved;
-    }
+        address uniswapInputToken = market.oracles[_token].inputToken;
 
-    function getCurrentValue(uint256 _requestId)
-        public
-        view
-        returns (
-            bool _ifRetrieve,
-            uint256 _value,
-            uint256 _timestampRetrieved
-        )
-    {
-        uint256 _count = oracle.getNewValueCountbyRequestId(_requestId);
-        _timestampRetrieved = oracle.getTimestampbyRequestIDandIndex(
-            _requestId,
-            _count - 1
+        uint256 value = OracleLibrary.getQuoteAtTick(
+            tick,
+            10 ** (IERC20(uniswapInputToken).decimals() - 1),
+            uniswapInputToken,
+            market.oracles[_token].outputToken
         );
-        _value = oracle.retrieveData(_requestId, _timestampRetrieved);
 
-        if (_value > 0) return (true, _value, _timestampRetrieved);
-        return (false, 0, _timestampRetrieved);
+        market.oracles[_token].usdPrice = value;
+        market.oracles[_token].lastUpdatedAt = now;
```

Comment:
This can be removed, we don't need to do this, event tracking from the contracts we use will persist this information